src/libs/antares/study/parts/short-term-storage/series.cpp

```diff
@@ -180,4 +179,87 @@ bool Series::validateLowerRuleCurve() const
     return checkVectBetweenZeroOne(maxInjectionModulation, "lower rule curve");
 }
 
+bool Series::validateInflowsSums(unsigned int firstHourOfTheWeek, unsigned int cycleDuration) const
+{
+    for (unsigned int cycleHour = 0; cycleHour < Antares::Constants::nbHoursInAWeek;
+            cycleHour += cycleDuration)
+    {
+        double sumInflows = 0.0;
+        double sumInjection = 0.0;
+        double sumWithdrawal = 0.0;
+
+        // sum until end of cycle or end of week
+        for (unsigned int i = 0;
+                i < cycleDuration && i + cycleHour < Antares::Constants::nbHoursInAWeek; i++)
+        {
+            unsigned int calendarHour = firstHourOfTheWeek + cycleHour + i;
+
+            sumInflows += inflows[calendarHour];
+            sumInjection += maxInjectionModulation[calendarHour];
+            sumWithdrawal += maxWithdrawalModulation[calendarHour];
+        }
+
+        if (sumInjection > sumInflows || sumWithdrawal < sumInflows )
+        {
+            logs.warning() << "Error at week: " << firstHourOfTheWeek + 1 << " for sts series "
+                "inflows, sums at cycle timesteps are wrong";
+
+            return false;
+        }
+    }
+    return true;
+}
+
+bool Series::validateCycle(unsigned int firstHourOfTheWeek, std::optional<double> initialLevel,
+        unsigned int cycleDuration) const
+{
+    // Check that the initial level is inside the rule curves at all cycle timesteps
+    if (initialLevel.has_value())
+    {
+        for (unsigned int hour = firstHourOfTheWeek;
+                hour < firstHourOfTheWeek + Antares::Constants::nbHoursInAWeek;
+                hour += cycleDuration)
+        {
+            if (upperRuleCurve[hour] < initialLevel || lowerRuleCurve[hour] > initialLevel)
```

Comment:
You could have 2 checks to have more specific logs (error in lower rule curve, error in upper rule curve)

Also, make sure to log "short-term storage", not sts